presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
+example::
+
+    presto> CREATE TABLE vals AS SELECT CAST('A' as CHAR(2001)) col;
+    presto> INSERT INTO vals (col) VALUES (CAST('BB' as CHAR(2001)));
+    presto> SELECT LENGTH(col) FROM vals;
+    2001
+    1
+
+Attempting to write a ``CHAR`` that doesn't fit in the column's actual size
+fails. This is also true for the equivalent ``VARCHAR`` types.
+
+
+Type mapping configuration properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table:: Type Mapping Properties
+  :widths: 20, 20, 50, 10
+  :header-rows: 1
+
+  * - Configuration property name
+    - Session property name
+    - Description
+    - Default
+  * - ``unsupported-type.handling-strategy``
+    - ``unsupported_type_handling_strategy``
+    - Configures how unsupported column data types are handled:
+
+      - ``IGNORE`` - column is not accessible.
+      - ``CONVERT_TO_VARCHAR`` - column is converted to unbounded ``VARCHAR``.
+
+    - ``IGNORE``
+  * - ``oracle.number.default-scale``
+    - ``number_default_scale``               
+    - Default Presto ``DECIMAL`` scale for Oracle ``NUMBER`` (without precision 
+      and scale) date type. When not set then such column is treated as not 
+      supported.
+    - not set
+  * - ``oracle.number.rounding-mode``
+    - ``number_rounding_mode``
+    - Rounding mode for the Oracle ``NUMBER`` data type. This is useful when 
+      Oracle ``NUMBER`` data type specifies higher scale than is supported in 
+      Presto. Possible values are:
+
+      - ``UNNECESSARY`` - Rounding mode to assert that the 
+        requested operation has an exact result, 
+        hence no rounding is necessary.
+      - ``CEILING`` - Rounding mode to round towards
+        positive infinity.
+      - ``FLOOR`` - Rounding mode to round towards negative
+        infinity.
+      - ``HALF_DOWN`` - Rounding mode to round towards
+        ``nearest neighbor`` unless both neighbors are
+        equidistant, in which case rounding down is used.
+      - ``HALF_EVEN`` - Rounding mode to round towards the
+        ``nearest neighbor`` unless both neighbors are equidistant,
+        in which case rounding towards the even neighbor is
+        performed.
+      - ``HALF_UP`` - Rounding mode to round towards
+        ``nearest neighbor`` unless both neighbors are
+        equidistant, in which case rounding up is used
+      - ``UP`` - Rounding mode to round towards zero.
+      - ``DOWN`` - Rounding mode to round towards zero.
+
+    - ``UNNECESSARY``
+
+Synonyms
+--------
+
+Based on performance reasons, Presto disables support for Oracle ``SYNONYM``. To
+include ``SYNONYM``, add the following: 
+
+.. code-block:: none
+
+    oracle.synonyms.enabled=true
+
+Pushdown
+--------
+
+The connector supports :doc:`pushdown </optimizer/pushdown>` for optimized query
+processing.
 
 Limitations
 -----------
 
-The following SQL statements are not yet supported:
+The following SQL statements aren't supported:
```

Comment:
Using a contraction here looks weird, since this is a formal statement. Contractions are fine in prose, but for a short single statement like this, I don't see a benefit to using it.

Reply:
I was using contractions based on the Google Style Guide: https://developers.google.com/style/contractions

> Negation contractions
In particular, it's fine to use -n't contractions, such as isn't, don't, and can't.

One reason that such contractions are useful is that it's sometimes easy for a reader to miss the word **not**, whereas it's harder to misread don't as do.